src/Elmish.WPF.Tests/ViewModelTests.fs

```diff
@@ -1195,11 +1196,11 @@ module SubModel =
 
 module SubModelSeq =
 
-  let private testObservableCollectionContainsExpectedItems (vm: ViewModel<_, _>) name expected =
+  let private testObservableCollectionContainsExpectedItems (vm: ViewModel<Guid list, (Guid * obj)>) name expected =
     let actual =
       vm.Get name
-      |> unbox<ObservableCollection<ViewModel<_,_>>>
-      |> Seq.map (fun vm -> vm.CurrentModel |> unbox)
+      |> unbox<ObservableCollection<obj>>
+      |> Seq.map (fun vm -> vm |> unbox<ViewModel<Guid,obj>> |> (fun vm -> vm.CurrentModel) |> unbox)
```

Comment:
No, because it's actually an `ObservableCollection<obj>` as its type. It's not until you actually get the items out of the type that you can then unbox them to be `ViewModel<Guid, obj>`. It's equivalent to linq's `.Cast<T>()` function. Just because you can `.Cast<T>()` something doesn't mean you can cast the collection to that type (exception being with covariant types like `IEnumerable<T>`)

Reply:
See my [comments here in context about changes to boxing](https://github.com/elmish/Elmish.WPF/pull/449/files#r834897266)